### PR TITLE
Updates for v0.8

### PR DESCRIPTION
**Pull request by dhardy**
I updated a few things from #28. @vks review?

### CODE REVIEW

**Comment by dhardy** [comment]
Note that the v0.8 update guide is [not currently live](https://rust-random.github.io/book/update.html) due to omission from `SUMMARY.md`. It should be good to go after this.

**Comment by dhardy** [comment]
BTW I hope you weren't also working on this. I assumed you weren't since I hadn't heard from you in a few days.

**Comment by vks** [comment]
I wasn't working on it, I was busy handing in my dissertation. 😅 But now that this is done, I have more time available again.

File: src/SUMMARY.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
     - [Updating to 0.5](update-0.5.md)
     - [Updating to 0.6](update-0.6.md)
     - [Updating to 0.7](update-0.7.md)
+    - [Updating to 0.8](update-0.8.md)
```
Comment (vks) [inline src/SUMMARY.md:27]: Why is the indentation different than for the other lines?

File: src/update-0.8.md [hunk 2]
```diff
@@ -49,44 +57,60 @@ let a: u32 = Standard.sample(&mut rng);
...
 
-[`Rng::gen_range`] now takes a `Range` instead of two numbers. This requires
-replacing `gen_range(a, b)` with `gen_range(a..b)` in code written for `rand
-0.7`. We suggest to replace the regular expression
-`gen_range\(([^,]*),\s*([^)]*)\)` with `gen_range(\1..\2)` (or
-`gen_range($1..$2)` in some tools without support for backreferences).
-Additionally, if `a` or `b` were a reference, explicit dereferencing may now be
-required. Inclusive ranges are now supported: `gen_range(a, b + 1)` can be
-replaced with `gen_range(a..=b)`.
+#### gen_range
```
Comment (vks) [inline src/update-0.8.md:60]: Maybe `gen_range` is better?

File: src/update-0.8.md [hunk 3]
```diff
@@ -49,44 +57,60 @@ let a: u32 = Standard.sample(&mut rng);
...
+Most IDEs support search-replace-across-files or similar; alternatively an
+external tool such as Regexxer may be used.
+
+This change has a couple of other implications:
+
+-   inclusive ranges are now supported, e.g. `gen_range(1..=6)` or `gen_range('A'..='Z')`
+-   it may be necessary to explicitly dereference some parameters
+-   SIMD types are no longer supported (`Uniform` types may still be used directly)
+
+#### fill
```
Comment (vks) [inline src/update-0.8.md:79]: Same as the other heading above.

File: src/update-0.8.md [hunk 4]
```diff
@@ -49,44 +57,60 @@ let a: u32 = Standard.sample(&mut rng);
...
-are supported by [`Rng::fill`] and [`Rng::try_fill`]. Now, the [`Fill`] trait
-has to be implemented instead: Rather than providing a mutable byte slice, the
-user-defined type must be filled with random data.
+only affect code implementing `AsByteSliceMut` on user-defined types, since the
+[`Rng::fill`] and [`Rng::try_fill`] retain support for previously-supported types.
+
+`Fill` supports some additional slice types which could not be supported with
+`AsByteSliceMut`: `[bool], [char], [f32], [f64]`.
+
+#### adapter
```
Comment (vks) [inline src/update-0.8.md:88]: See above.

**Review by vks — APPROVED** [review]

File: src/SUMMARY.md (hunk 1 above)
Comment (dhardy) [reply src/SUMMARY.md:27]: It's not?

File: src/update-0.8.md (hunk 2 above)
Comment (dhardy) [reply src/update-0.8.md]: I didn't realise in-line code was supported in titles (don't think it is on GH). Works here though.

**Comment by dhardy** [comment]
Well, congratulations! Nice to get that done I'm sure.

File: src/SUMMARY.md (hunk 1 above)
Comment (vks) [reply src/SUMMARY.md:27]: Sorry, seems to be a bug with the GitHub app.

**Comment by vks** [comment]
Thanks!

I think this is good to go.

**Pull request merged**